modules/@apostrophecms/doc-type/index.js

```diff
@@ -0,0 +1,25 @@
+module.exports = {
+  options: {
+    forbiddenSlugs: [
+      '/evil-page',
+      'evil-piece'
+    ]
+  },
+  handlers(self) {
+    return {
+      beforeSave: {
+        checkForbiddenSlugs(req, doc) {
+          if (self.options.forbiddenSlugs.includes(doc.slug)) {
+            const e = self.apos.error('invalid', 'That slug is reserved.');
+            e.path = 'slug';
+            throw self.apos.error('invalid', {
+              errors: [
+                e
```

Comment:
Including it in an `errors` array in the final error object causes the Apostrophe UI to loop through and find it and display it in the right place, along with any errors you wish to report on other fields.